src/components/React.js

```diff
@@ -51,7 +51,8 @@ class React {
 
         this.options = Object.assign(
             {
-                extractStyles: false
+                extractStyles: false,
+                localIdentName: '[name]_[local]__[hash:base64:5]'
```

Comment:
ideally we should follow what is happening for Vue, https://github.com/aarbi/laravel-mix/blob/e37bbba53bf72d3fbb398556e4a5e3edba4eb91c/src/components/Vue.js#L22
extractStyles accept both boolean and string, and the file name would be provided to it. So let's use that instead

Reply:
localIdentName has nothing to do with extractStyles property.  Please see the following link: https://docs.w3cub.com/webpack/loaders/css-loader#localidentname

Reply:
I guess I understand your point now, this is a bit off the goal of my branch though.
my PR was a WIP though, it is not working yet fully. It would be better if you start a new PR against the original upstream then!

Reply:
It's your decision, I shared code and since it depends on your purposed changes I shared the fixed code of extractStyles and added localIdentName ability as well. I am already using my shared code on my project as a mix module.